### DISCUSSION TITLE
Possibility of loading different .NET frameworks

### DISCUSSION THREAD

**Discussion by mflova** [pythonnet/pythonnet#2484] [Q&A]
I have an application that requires me to load both .NET Framework and .NET Core. However, as soon as you `import clr` Python creates a "singleton" where further imports within the code will load the cached module. Therefore, I only managed to load one of them:

```
import importlib
clr = importlib.import_module("clr")  # This one will load clr using .NET Framework


from pythonnet import load
load("coreclr")
clr2 = importlib.import_module("clr")  # This one will still use .NET Framework as the module was cached
```

Is there a possibility of loading two versions of `clr`, each one with .NET Framework and .NET Core respectively?

**Comment by sschukat** [comment]
Even if it would be possible, multiple runtimes in one process is stated as  _The official answer is that it is not supported and untested and therefore there are no guarantees._ [Issue 53729](https://github.com/dotnet/runtime/issues/53729#issuecomment-895604060) from a core developer of .NET. For the special use case of multiple .NET core version there is a deeper technical explanation in [Issue 12018](https://github.com/dotnet/runtime/issues/12018#issuecomment-551214265) by the same developer. Hence I would suggest to analyze other possibilities for your use case. In our software we make a clear process separation.
